signature/mechanism.go

```diff
@@ -34,6 +35,21 @@ type SigningMechanism interface {
 	UntrustedSignatureContents(untrustedSignature []byte) (untrustedContents []byte, shortKeyIdentifier string, err error)
 }
 
+// SigstoreSigningMechanism abstracts a way to sign binary blobs and verify their signatures.
+// Each mechanism should eventually be closed by calling Close().
+// TODO: consolidate with SigningMechanism interface above.
+type SigstoreSigningMechanism interface {
+	// Close removes resources associated with the mechanism, if any.
+	Close() error
+	// SupportsSigning returns nil if the mechanism supports signing, or a SigningNotSupportedError.
+	SupportsSigning() error
+	GenerateCertificate() error
+	// Sign creates a (non-detached) signature of input using ephemeral keys.
+	// Fails with a SigningNotSupportedError if the mechanism does not support signing.
+	Sign(payload []byte) (signature []byte, err error)
+	Upload(signature, payload []byte) (*models.LogEntryAnon, error)
```

Comment:
Should this be public at all? See elsewhere.

Reply:
`internal` is an interesting point — it didn’t even exist when `c/image/signature` was first written.

OTOH we might want to expose _something_ lower-level than `copy.Image`; e.g. `podman image sign` exists (as much as I’d like for that not to be the case), and other callers might have their own production/publishing/… workflows that don’t want `copy.Image`, while we’d still prefer them to use a library for actual signing (and much more importantly, verification).

But, short-term, doing things in `internal` makes the API design bar much lower, and we can always delay adding a public API until after there is a known user/need.